api/src/main/java/jakarta/enterprise/lang/model/AnnotationInfo.java

```diff
@@ -18,15 +32,15 @@
     /**
      * Declaration of this annotation's type.
      *
-     * @return declaration of this annotation
+     * @return declaration of this annotation, never {@code null}
      */
-    ClassInfo<?> declaration();
+    ClassInfo<T> declaration();
 
     /**
      * Fully qualified name of this annotation.
      * Equivalent to {@code declaration().name()}.
      *
-     * @return fully qualified name of this annotation
+     * @return fully qualified name of this annotation, never {@code null}
```

Comment:
Off topic, since this was introduced earlier and you didn't change it, but I want to bring this up anyway.

I recently learned that the JLS makes an important distinction between fully qualified name and binary name. They are identical for top-level types, but not for nested types.

For example:

```java
package com.example;

public class MyClass {|
    public @interface MyAnnotation {
    }
}
```

Fully qualified name of the `MyAnnotation` type is `com.example.MyClass.MyAnnotation`, while binary name is `com.example.MyClass$MyAnnotation`.


I _think_ we should probably settle on binary names, and I started doing that here: https://github.com/eclipse-ee4j/cdi/pull/506/files#diff-a8c5a19048a48d991137c3ecb9f78f593b33a264b81a41bd5fa543cf9ac7ae73

WDYT about that?

Reply:
+1 for binary names